lib/Parser/RegexRuntime.h

```diff
@@ -706,7 +697,7 @@ namespace UnifiedRegex
 
     struct Inst : protected Chars<char16>
     {
-        enum InstTag : uint32
+        enum InstTag : uint8
```

Comment:
It's hard to see the benefits of this because there is a vtable in test builds.
see `virtual int Print(DebugWriter*w, Label label, const Char* litbuf) const = 0;`

It looks like in test build, the size remains 12 bytes (8 for vtable and 4 for tag) even with uint8.
However, in release build (and if you comment out the virtual method) the size goes from 4 bytes to 1 byte.

I wonder if the print function should use a similar mechanism to how we invoke `Exec`, ie: a switch on `tag` and a cast on the `inst`.

it would bring our test build closer to what we ship and also represent better the impact of such a change

Reply:
The vtable is gone because I removed virtual from Print and do manual dispatch (debug only) in #4066 which this PR is based on (for that reason).